### PR TITLE
Add output_format and handle_when parameters for checks

### PR DESCRIPTION
**Pull request by treydock**
# Pull Request Checklist

## Description

Add `output_format` and `handle_when` parameters for checks.

https://docs.sensu.io/sensu-enterprise/latest/built-in-mutators/
https://docs.sensu.io/sensu-enterprise/latest/filters/handle-when/

### CODE REVIEW

File: manifests/check.pp [hunk 1]
```diff
@@ -130,6 +138,9 @@
   Variant[Undef,Enum['absent'],Hash] $subdue = undef,
   Variant[Undef,Enum['absent'],Hash] $proxy_requests = undef,
   Variant[Undef,Enum['absent'],Hash] $hooks = undef,
+  Optional[Enum['influxdb_line', 'graphite_plaintext', 'nagios_perfdata',
```
Comment (ghoneycutt) [inline manifests/check.pp]: Should this be `String[1]`? Could you have custom output formats? @asachs01

Comment (treydock) [reply manifests/check.pp]: According to this: https://docs.sensu.io/sensu-enterprise/latest/built-in-mutators/ those values are the only ones allowed, I would think custom is not allowed as that would likely require custom code being a part of Sensu.

Comment (ghoneycutt) [reply manifests/check.pp]: Aaron let me know that you could write your own mutator, so let's keep it broad.

Comment (treydock) [reply manifests/check.pp]: I updated to just require a string.

**Comment by treydock** [comment]
Had to add several fixes specific to testing. These are things we also had to patch with sensu-puppet module.

**Pull request merged**